include/pybind11/pybind11.h

```diff
@@ -51,6 +51,7 @@ NAMESPACE_BEGIN(PYBIND11_NAMESPACE)
 class cpp_function : public function {
 public:
     cpp_function() { }
+    cpp_function(std::nullptr_t) { }
```

Comment:
This is pretty minor, but I think with this `nullptr` constructor it would be slightly nicer to change `def_property_readonly` and `def_property_readonly_static` to pass `nullptr` rather than `cpp_function()`.  It won't make any difference to how it works, but conceptually it seems nicer since the PR adds (and documents) passing a `nullptr`.

Reply:
Related to this: it should also be possible to specify a default-constructed `cpp_function()` for either the setter or getter, so perhaps add another property such as:

```C++
.def_property("impossible", py::cpp_function(), py::cpp_function())
```
with associated tests for attribute errors on *both* set or get (as per my other comment), but mainly just so we still have some test suite code invoking that code path.